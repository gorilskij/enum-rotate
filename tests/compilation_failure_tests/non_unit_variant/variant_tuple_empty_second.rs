use derive_enum_rotate::EnumRotate;

#[derive(EnumRotate)]
enum Enum {
    A,
    B(),
    C,
}
