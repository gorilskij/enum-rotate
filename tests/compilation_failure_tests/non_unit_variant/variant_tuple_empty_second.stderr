error: Variant B is not a unit variant, all variants must be unit variants to derive EnumRotate
 --> tests/compilation_failure_tests/non_unit_variant/variant_tuple_empty_second.rs:6:5
  |
6 |     B(),
  |     ^^^

error[E0601]: `main` function not found in crate `$CRATE`
 --> tests/compilation_failure_tests/non_unit_variant/variant_tuple_empty_second.rs:8:2
  |
8 | }
  |  ^ consider adding a `main` function to `$DIR/tests/compilation_failure_tests/non_unit_variant/variant_tuple_empty_second.rs`
